internal/store/leaf/fsck.go

```diff
@@ -180,36 +245,36 @@ type convertedSecret interface {
 	FromMime() bool
 }
 
-func (s *Store) fsckCheckEntry(ctx context.Context, name string) (string, ErrorSeverity, error) {
-	errcoll := ""
+func (s *Store) fsckCheckEntry(ctx context.Context, name string) (string, error) {
+	errs := &fsckMultiError{}
 	recpNeedFix := false
 
-	errs, err := s.fsckCheckRecipients(ctx, name)
-	if err != nil {
-		if errs == errsFatal {
-			return "", errsFatal, fmt.Errorf("Checking recipients for %s failed:\n    %w", name, err)
+	merr := s.fsckCheckRecipients(ctx, name)
+	if merr.ErrorOrNil() != nil {
+		if merr.Severity == errsFatal {
+			return "", errs.Append(errsFatal, fmt.Errorf("Checking recipients for %s failed:\n    %w", name, merr)).ErrorOrNil()
 		}
 		// the only errsNonFatal error from that function are missing/extra recipients,
 		// which isn't much of an error since we have yet to correct that.
 		recpNeedFix = true
-		errcoll += err.Error()
+		_ = errs.Append(merr.Severity, merr)
 	}
 
 	// make sure we are actually allowed to decode this secret
 	// if this fails there is no way we could fix anything
 	if !IsFsckDecrypt(ctx) {
 		if !recpNeedFix {
-			return "", errsNil, nil
+			return "", nil
 		}
 
-		return "", errsFatal, fmt.Errorf(errcoll + "\nRun fsck with the --decrypt flag to re-encrypt it automatically, or edit this secret yourself.")
+		return "", errs.Append(errsFatal, fmt.Errorf("Run fsck with the --decrypt flag to re-encrypt it automatically, or edit the secret %s yourself.", name)).ErrorOrNil()
```

Comment:
I'm a bit puzzled here.
If we're not running with `--decrypt`, we would get one error and abort because of the `errsFatal` at the first entry with missing recipients, right? 
Or maybe worse, one per entry.

But then we don't continue and so we don't know for instance if there are MIME leftovers, unless we're running with `--decrypt`, right?

Reply:
Yes, I think so. But seriously: This code is a mess. It's been like that since I first wrote it. We're making baby steps to improve it, but not by much. Eventually we'll get there ... (but that will need more untangling and reporting changes).